compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1392,6 +1393,13 @@ rustc_queries! {
         feedable
     }
 
+    /// List of autodiff extern functions in the current crate.
+    query autodiff_attrs(def_id: DefId) -> &'tcx AutoDiffAttrs {
+        desc { |tcx| "computing autodiff attributes of `{}`", tcx.def_path_str(def_id) }
+        arena_cache
+        cache_on_disk_if { def_id.is_local() }
+    }
+
```

Comment:
So I gave it some thought why your attributes are disappearing across crates. You'd have to add `separate_provide_extern` to make that work, but there's a simpler way: just add `AutoDiffAttrs` to `CodegenFnAttrs`, which already handles all that (and logically I'd think it would belong there anyway?)

Reply:
That should also fix the ICE in CI